### DISCUSSION TITLE
CloudMonitoring:  Add support for new operators

### DISCUSSION THREAD

**Discussion by whuut** [grafana/grafana#41822] [Data Sources]
Hello,

I want to filter on a metric with a numeric value in the stackdriver data source.
More precisely the metric `appengine.googleapis.com/http/server/response_count`, which is of type INT64, see [1].

The stackdriver data source only offers 4 different operators (`=, !=, =~, !=~`), which expand to either the `start_with` function or the `monitoring.regex.full_match` function, which both are only valid for values of type string.

That's why the stackdriver data source raises the following error if I want to filter on numeric values (e.g. all response codes greater than 400):

```Field filter had an invalid value of "metric.type="appengine.googleapis.com/http/server/response_count" resource.label.module_id="default" metric.label.response_code=starts_with("4")": Function restrictions are only allowed for string type.```

This happens with `= 4*` or any variation of `=~ 4\d*`.

There are other operators available in the Monitoring v3 API [2], but the stackdriver data source does not let me put in custom operators in the operator field. For example, if I put in the operator `>` which would probably work for numeric values, it just resets the operator field to blank.

So it is not possible to filter on numeric values at the moment, but the data source documentation [3] does not mention this limitation.

[1] https://cloud.google.com/monitoring/api/metrics_gcp
[2] https://cloud.google.com/monitoring/api/v3/sorting-and-filtering#filter_syntax
[3] https://grafana.com/docs/grafana/v6.5/features/datasources/stackdriver/

**Environment**:
- Grafana version: 6.6.2
- Data source type & version: Stackdriver

**Comment by andresmgot** [comment]
I believe this should no longer be an issue. I can succesfully use regexes in a filter:

![Screenshot from 2022-11-07 17-48-12](https://user-images.githubusercontent.com/4025665/200367490-d0d8e893-9ea3-4346-ba6e-996db1efad4a.png)

I am archiving this discussion but if you find any problem with this, please open a new issue.
